[LT] getBackend returns non-const pointer

alanwaketan opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #76521
* #76517

Summary:
Let's change getBackend() to a return non-const pointer such that
we can call setters.

Test Plan:
./build/bin/test_lazy

Differential Revision: [D35994891](https://our.internmc.facebook.com/intern/diff/D35994891)

Commit:
[LT] getBackend returns non-const pointer

Summary:
Let's change getBackend() to a return non-const pointer such that
we can call setters.

Test Plan:
./build/bin/test_lazy

[ghstack-poisoned]

alanwaketan commented:
@alanwaketan has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D35994891).

wconstab commented on torch/csrc/lazy/ts_backend/ts_backend_impl.cpp:
```diff
@@ -145,7 +145,7 @@ class TSBackendImpl : public torch::lazy::BackendImplInterface {
   torch::lazy::BackendDevice GetBackendDevice(
       c10::Device device) const override;
 
-  void SetRngSeed(size_t seed) const override {
```
oh yea, I remember noticing this at one point and wanting to fix it.

alanwaketan commented:
Thanks Will for approving this.

Commit:
Update on "[LT] getBackend returns non-const pointer"


Summary:
Let's change getBackend() to a return non-const pointer such that
we can call setters.

Test Plan:
./build/bin/test_lazy

Differential Revision: [D35994891](https://our.internmc.facebook.com/intern/diff/D35994891)

[ghstack-poisoned]

Commit:
Update on "[LT] getBackend returns non-const pointer"


Summary:
Let's change getBackend() to a return non-const pointer such that
we can call setters.

Test Plan:
./build/bin/test_lazy

Differential Revision: [D35994891](https://our.internmc.facebook.com/intern/diff/D35994891)

[ghstack-poisoned]

Commit:
Update on "[LT] getBackend returns non-const pointer"


Summary:
Let's change getBackend() to a return non-const pointer such that
we can call setters.

Test Plan:
./build/bin/test_lazy

Differential Revision: [D35994891](https://our.internmc.facebook.com/intern/diff/D35994891)

[ghstack-poisoned]

Commit:
Update on "[LT] getBackend returns non-const pointer"


Summary:
Let's change getBackend() to a return non-const pointer such that
we can call setters.

Test Plan:
./build/bin/test_lazy

Differential Revision: [D35994891](https://our.internmc.facebook.com/intern/diff/D35994891)

[ghstack-poisoned]

Commit:
Update on "[LT] getBackend returns non-const pointer"


Summary:
Let's change getBackend() to a return non-const pointer such that
we can call setters.

Test Plan:
./build/bin/test_lazy

Differential Revision: [D35994891](https://our.internmc.facebook.com/intern/diff/D35994891)

[ghstack-poisoned]

Krovatkin committed:
Update on "[LT] getBackend returns non-const pointer"


Summary:
Let's change getBackend() to a return non-const pointer such that
we can call setters.

Test Plan:
./build/bin/test_lazy

Differential Revision: [D35994891](https://our.internmc.facebook.com/intern/diff/D35994891)

[ghstack-poisoned]

Krovatkin committed:
Update on "[LT] getBackend returns non-const pointer"


Summary:
Let's change getBackend() to a return non-const pointer such that
we can call setters.

Test Plan:
./build/bin/test_lazy

Differential Revision: [D35994891](https://our.internmc.facebook.com/intern/diff/D35994891)

[ghstack-poisoned]
